packages/picasso.js/src/core/dock-layout/dock-layout.js

```diff
@@ -173,11 +224,13 @@ function positionComponents(components, logicalContainerRect, reducedRect, conta
   const referencedComponents = {};
   const referenceArray = components.slice();
   components.sort((a, b) => {
-    if (/^@/.test(b.config.dock())) {
-      return -1;
-    }
-    if (/^@/.test(a.config.dock())) {
-      return 1;
+    if (a.referencedDocks.length > 0 || b.referencedDocks.length > 0) {
```

Comment:
No need add an extra if check and than use the "old" reference check. Just do:
```js
if (b.referencedDocks.length > 0) {
  return -1;
}
if (a.referencedDocks.length > 0) {
  return 1;
}
```

Reply:
Good point! Thanks!

Reply:
fixed!